### PR TITLE
vkvideoparser: Export CreateVulkanVideoParser with C++ mangling.

### PR DESCRIPTION
**Pull request by charlie-ht**
Due to the Vulkan types being behind a namespace in the CTS, for direct linking, it's required that we export the symbol without demangling, otherwise there's no convenient way to match the argument types exposed by the GStreamer-side API.

Going forward we can work on making this interface nicer.

### CODE REVIEW

**Review by ceyusa — APPROVED** [review]
lgtm

**Comment by ceyusa** [comment]
mmh... it breaks test compilation

**Comment by charlie-ht** [comment]
I went ahead and started removing the dlopen cases, perhaps we want to also remove the external_libs in this PR as well.

File: test/meson.build [hunk 1]
```diff
@@ -7,29 +7,8 @@ gsttest = executable(
   override_options: _override_options,
 )
 
-gstdlopentest = executable(
```
Comment (dabrain34) [inline test/meson.build:10]: you can remove the sources_dlopen including the files.

**Comment by charlie-ht** [comment]
Let's pause this line of work. @dabrain34 let me know about some other issues like: https://github.com/Igalia/GstVkVideoParser/issues/8

I'll take a look at some of this corner cases before droppign the ability to use Nvidia's parser.

The test failure is a bit weird, didn't reproduce locally.
